simulator/infra/replacement/cache_replacement.h

```diff
@@ -0,0 +1,48 @@
+/**
+ * CacheReplacement.h
+ * Header for the Cache replacement algorithms module.
+ * @author Oleg Ladin, Denis Los, Andrey Agrachev
+ */
+
+#ifndef CACHEREPLACEMENT_H
+#define CACHEREPLACEMENT_H
+
+#include <list>
+#include <unordered_map>
+
+enum replacement_polices { LRU, pseudo_LRU };
+
+struct CacheReplacementInterface
+{
+    explicit CacheReplacementInterface() {}
+    virtual ~CacheReplacementInterface() {}
+    virtual void touch( std::size_t) = 0;
+    virtual void set_to_erase( std::size_t) = 0;
+    virtual void allocate( std::size_t) = 0;
+    virtual std::size_t update() = 0;
+    virtual std::size_t get_ways() const = 0;
+    virtual std::size_t get_hash_size() const = 0;
+};
+
```

Comment:
I suggest to put a factory method here:
```c++
std::unique_ptr<CacheReplacementInterface> create_cache_replacement( const std::string& name, std::size_t ways);
```

Then you can put definition of LRUCacheInfo class into `cache_replacement.cpp`, leaving only the interface visible outside.